김현국/app/src/main/java/com/example/myapplication/adapter/RepoAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.myapplication.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.myapplication.databinding.ItemSearchBinding
+import com.example.myapplication.model.Repository
+import com.example.myapplication.presenter.search.OnItemClick
+import com.example.myapplication.presenter.search.SearchAdapterContract
+
+/**
+ * @author 김현국
+ * @created 2022/04/25
+ */
+class RepoAdapter() :
+    RecyclerView.Adapter<RepoAdapter.ViewHolder>(),
+    SearchAdapterContract.View,
+    SearchAdapterContract.Model {
+
+    private var repos: List<Repository>? = emptyList()
+    private var onItemClick: OnItemClick? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding = ItemSearchBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(repos!![position])
+        holder.itemView.setOnClickListener {
+            onItemClick!!.onItemClick(position)
+        }
+    }
+
+    override fun getItemCount(): Int {
+        return repos!!.size
+    }
+
+    class ViewHolder(private val binding: ItemSearchBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(repo: Repository) {
+            binding.tvMainName.text = repo.name
+            if (repo.language != null) {
+                binding.tvMainLanguage.text = repo.language
+            }
```

Comment:
이 부분은 의도하신 부분과 다를 수 있어서 참고만 해주세요!
TextView.setText()에는 null 값이 들어갈 수도 있어서 null-check을 안해주기도 하더라구요
```suggestion
            if (repo.language != null) {
                binding.tvMainLanguage.text = repo.language
            }
```

Reply:
아이구 suggestion을 수정안하고 코멘트를 달아서 다시 수정했습니다~ ㅋㅋㅋㅋ